chunks/chunks.go

```diff
@@ -53,62 +57,30 @@ func (cm *ChunkMeta) writeHash(h hash.Hash) error {
 	return nil
 }
 
-// deletedIterator wraps an Iterator and makes sure any deleted metrics are not
-// returned.
-type deletedIterator struct {
-	it chunks.Iterator
-
-	intervals Intervals
-}
-
-func (it *deletedIterator) At() (int64, float64) {
-	return it.it.At()
-}
-
-func (it *deletedIterator) Next() bool {
-Outer:
-	for it.it.Next() {
-		ts, _ := it.it.At()
-
-		for _, tr := range it.intervals {
-			if tr.inBounds(ts) {
-				continue Outer
-			}
-
-			if ts > tr.Maxt {
-				it.intervals = it.intervals[1:]
-				continue
-			}
-
-			return true
-		}
-
-		return true
-	}
+var (
+	errInvalidSize     = fmt.Errorf("invalid size")
+	errInvalidFlag     = fmt.Errorf("invalid flag")
+	errInvalidChecksum = fmt.Errorf("invalid checksum")
+)
 
-	return false
-}
+// The table gets initialized with sync.Once but may still cause a race
```

Comment:
This comment relates rather to init body, right? Can we move it there?

Reply:
Comment needs to be updated.